src/fe-common/core/chat-completion.c

```diff
@@ -639,6 +639,29 @@ static void complete_window_nicks(GList **list, WINDOW_REC *window,
         }
 }
 
+/* iterates over all nicks in the list and escapes them if needed */
+static void escape_nicks(GList **list)
+{
+	GList *l;
+	char *escaped;
+
+	/* verify that the list isn't NULL */
+	g_return_if_fail(list != NULL);
+
+	/* iterate over the nick list */
+	for (l = *list; l != NULL; l = l->next) {
+		/* only escape if we find either \\r or \\n in the nick */
+		if (l->data != NULL &&
+		    (strstr(l->data, "\\r") != NULL ||
+		     strstr(l->data, "\\n") != NULL)) {
```

Comment:
`expand_escapes`/`expand_escape` handle much more than just `\r` and `\n`, there's \xFF, `\cA`, `\777` and so on.